### PR TITLE
Added options for generate parameter file

### LINKED ISSUE #25: Additional functionality to Build-Bicep -GenerateParameterFile
**Is your feature request related to a problem? Please describe.**
The `-GenerateParameterFile` switch generates a parameter file with all parameters in the `.bicep` file.  You should be able to specify what type of parameters should be added to the parameter file.

**Describe the solution you'd like**
`-GenerateParameterFile` should be a parameter supporting the following scenarios.

`-GenerateParameterFile All` - Should generate a parameter file with all parameters from template (Parameters with expressions should have expression as value)

`-GenerateParameterFile Required` - Should generate a parameter file with required parameters only

`-GenerateParameterFile Default` - Should generate a parameter file with all parameters that does not have an expression as default value

**Additional context**

### PR DESCRIPTION
**Pull request by StefanIvemo**
This PR partially closes #25. But I'm not sure if we need the `-Default` option mentioned in the issue or not. 

- Replaced the `-GenerateParameterFile` switch with two new switches in ´Build-Bicep`
  - `-GenerateAllParametersFile` generates a parameter file with all parameters in the bicep file
  - `-GenerateRequiredParametersFile` generates a parameter file with the required parameters only
  - Since we already have six parameter sets for `Build-Bicep` I've added a check if someone uses both switches at the same time and will then choose `All` and write a warning instead of adding another parameter set.

- Added the parameter `-Parameters` to `New-BicepParameterFile`, accepts inputs `All` or `Required
  - `All` generates a parameter file with all parameters in the bicep file
  - `Required` generates a parameter file with the required parameters only
  - If the parameter isn't provided it will default to `Required`

- Updated help files for both `Build-Bicep` and `New-BicepParameterFile`
- Updated external help file

How to use:

params.bicep
```bicep
param resourceName string
param rules object
param location string = resourceGroup().location
param accessTier string = 'SuperHot'
```

```powershell
#Generate parameter file during build with all parameters
Build-Bicep -Path params.bicep -GenerateAllParametersFile

#Generate parameter file during build with required parameters only
Build-Bicep -Path params.bicep -GenerateRequiredParametersFile

#Create parameter file with required parameters only
New-BicepParameterFile -Type params.bicep -Parameters Required
or
New-BicepParameterFile -Path params.bicep

#Create parameter file with all parameters
New-BicepParameterFile -Path params.bicep -Parameters All
```

### CODE REVIEW

**Comment by StefanIvemo** [comment]
Updated PR description after new commits, made a lot of changes.

File: Docs/Help/Build-Bicep.md [hunk 1]
```diff
@@ -219,6 +206,51 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+### -OutputPath
+{{ Fill OutputPath Description }}
```
Comment (bjompen) [inline Docs/Help/Build-Bicep.md:210]: Missing parameter description here :)

Comment (StefanIvemo) [reply Docs/Help/Build-Bicep.md]: Ah, nice.

**Review by emilguden — DISMISSED** [review]
Looks good!

**Review by emilguden — APPROVED** [review]
One more time :)

**Review by bjompen — APPROVED** [review]

**Pull request merged**